contracts/twap/CachedTwap.sol

```diff
@@ -18,20 +18,18 @@ abstract contract CachedTwap is CumulativeTwap {
         uint256 latestPrice,
         uint256 latestUpdatedTimestamp
     ) internal virtual returns (uint256) {
-        // if requested interval is not the same as the one we have cached, then call _getPrice() directly
+        // if the requested interval is not the same as the one we have cached, call _getTwap() directly
         if (_interval != interval) {
-            return _calculateTwapPrice(interval, latestPrice, latestUpdatedTimestamp);
+            return _getTwap(interval, latestPrice, latestUpdatedTimestamp);
         }
 
-        // if twap has been calculated in this block, then return cached value directly
-        if (_blockTimestamp() == _lastUpdatedAt) {
-            return _cachedTwap;
+        // only calculate twap and cache it when there's a new timestamp
+        if (_blockTimestamp() != _lastUpdatedAt) {
+            _update(latestPrice, latestUpdatedTimestamp);
```

Comment:
fount that our tx is failed here frequently, we might add another condition check here to prevent from reverting as below tx.
https://dashboard.tenderly.co/perpprotocol/curie-optimism/tx/optimistic/0xbf835dc0cdfaa70dc3a43d3aa0f41bd6371de4abcbfef200d14e504745e5867c/debugger?trace=0.2.3.2.0.0.2.0.3.0.2.0.0.1.1.0

@twxia